clustered/client/src/main/java/org/ehcache/clustered/client/internal/store/operations/PutIfAbsentOperation.java

```diff
@@ -32,10 +40,11 @@ public OperationCode getOpCode() {
    * for the same key.
    */
   @Override
-  public Operation<K> apply(final Operation<K> previousOperation) {
+  public Operation<K, V> apply(final Operation<K, V> previousOperation) {
     if(previousOperation == null) {
       return this;
     } else {
+      assertSameKey(previousOperation);
       return previousOperation;
```

Comment:
This is wrong I believe, because if the application of the previousOperation results in `null` then the `PutIfAbsentOperation` is applicable.